services/director-v2/src/simcore_service_director_v2/models/schemas/clusters.py

```diff
@@ -0,0 +1,27 @@
+from models_library.clusters import Cluster
+from models_library.generics import DictModel
+from pydantic import BaseModel
+from pydantic.networks import AnyUrl
+from pydantic.types import ByteSize, PositiveFloat
+
+
+class Worker(BaseModel):
+    id: str
+    name: str
+    resources: DictModel[str, PositiveFloat]
+    memory_limit: ByteSize
+
+
+class WorkersDict(DictModel[AnyUrl, Worker]):
```

Comment:
THOUTGH: This is a real model and not a ``TypedDict``, right? 

Up to know I would name a class derived from ``TypedDict`` with suffix ``Dict`` (analogous to suffix ``Str``, ...  ), in this case, if i did not see the full code, I  would have guessed that you refer to a "raw" python dict to keep the data
```python
from typing import TypedDict

class WorkersDict(TypedDict):
       id: str
       name: str
       resources: Dict[str,float]
       memory_limit: int
```

what you intent here is to have a dictionary with worker models as values, therfore, why not simply use directly ``DictModel[AnyUrl, Worker]`` ??  I think there is no need to add more variable names since it is pretty readable. Analogously we have ``Envelope[Worker]`` or ``Page[Worker]``. 

Conclusion, when you have a generic to represent a container or object wrapper, it is better to use it "as is" instead of creating a new variable.